### PR TITLE
Fix leaks

### PR DESCRIPTION
**Pull request by jschueller**
- fix lots of leaks reported by `-fsanitize=address,undefined`
- also downcast error from #2
- add cmake option for openmp (eases debugging)

### CODE REVIEW

**Comment by jschueller** [comment]
hello @Quentin62 , could you take a look ?

**Comment by Quentin62** [comment]
I will take a look this weekend

File: mixmodLib/SRC/mixmod/DiscriminantAnalysis/Predict/PredictInput.h [hunk 1]
```diff
@@ -93,7 +93,8 @@ class PredictInput : public Input
 
 protected:
 	/// pointer to a classification rule
-	Parameter *_classificationRule;
+	Parameter *_classificationRule = nullptr;
+	ParameterDescription * _parameterDescription = nullptr;
```
[the remark this answers is not in the document]
Comment (jschueller) [reply mixmodLib/SRC/mixmod/DiscriminantAnalysis/Predict/PredictInput.h:97]: PredictInput now takes ownership of the ParameterDescription pointer passed as parameter,
so it must store it delete it
thats one way to solve the leak from ExampleDataUtil

File: CMakeLists.txt [hunk 2]
```diff
@@ -77,7 +77,10 @@ message(STATUS "CMAKE_BUILD_TYPE : " ${CMAKE_BUILD_TYPE})
 # Compiler settings
 #-------------------
 set (CMAKE_CXX_STANDARD 17)
-find_package(OpenMP)
+option (MIXMOD_ENABLE_OPENMP "Enable OpenMP multithreading" ON)
```
[the remark this answers is not in the document]
Comment (jschueller) [reply CMakeLists.txt:80]: ok, done

**Review by Quentin62 — APPROVED** [review]

**Pull request merged**